DBZ-8318 Explain how to manage SPI and bundled jars for debezium engine applications

mfvitale opened the pull request:
closes: https://issues.redhat.com/browse/DBZ-8318

mfvitale commented:
PR for the FAQ https://github.com/debezium/debezium.github.io/pull/1075

roldanbob commented on documentation/modules/ROOT/pages/development/engine.adoc:
```diff
@@ -82,6 +82,30 @@ Or for the MongoDB connector:
+When an interface has multiple implementations — for instance, `io.debezium.snapshot.spi.SnapshotLock`, which has a default implementation in the core and specific ones for each connector — you must explicitly configure your build tools to merge the `META-INF/services files`.
+If this step is missed, Debezium may fail to locate the required implementation.
+
+To address this, instruct your build tools accordingly. For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
+you need to add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer:
```
```suggestion
For example, if you are using the https://maven.apache.org/plugins/maven-shade-plugin[Maven shade plugin],
add the https://maven.apache.org/plugins/maven-shade-plugin/examples/resource-transformers.html#ServicesResourceTransformer[`ServicesResourceTransformer`] transformer, as shown in the following example:
```

roldanbob commented on documentation/modules/ROOT/pages/development/engine.adoc:
```diff
@@ -82,6 +82,30 @@ Or for the MongoDB connector:
+</configuration>
+----
+
+Alternatively, if using the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in] you can utilize the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```
```suggestion
Alternatively, if you use the https://maven.apache.org/plugins/maven-assembly-plugin/index.html[Maven Assembly plug-in], you can use the https://maven.apache.org/plugins/maven-assembly-plugin/examples/single/using-container-descriptor-handlers.html[metaInf-services container descriptor handlers].
```

roldanbob reviewed (changes requested):
@mfvitale @jpechane I made a few suggestions. Let me know if you have any questions. Thanks!

mfvitale commented:
> @mfvitale @jpechane I made a few suggestions. Let me know if you have any questions. Thanks!

@roldanbob Thanks for your review. I have addressed your suggestions.

roldanbob reviewed (approved):
LGTM! Thanks!
